### DISCUSSION TITLE
Release 1.3.0

### DISCUSSION THREAD

**Discussion by TheR1D** [TheR1D/shell_gpt#471] [Announcements]
## What's Changed
* Ollama and other LLM backends.
* Markdown formatting now depends on role description.
* Code refactoring and optimisation.

### Multiple LLM backends
ShellGPT now can work with multiple Backends using [LiteLLM](https://github.com/BerriAI/litellm). You can use locally hosted open source models which are available for free. To use local models, you will need to run your own LLM backend server such as [Ollama](https://github.com/ollama/ollama). To setup ShellGPT with Ollama, please follow this comprehensive [guide](https://github.com/TheR1D/shell_gpt/wiki/Ollama). Full list of supported models and providers [here](https://docs.litellm.ai/docs/providers). **Note that ShellGPT is not optimized for local models and may not work as expected❗️**

### Markdown formatting
Markdown formatting now depends on the role description. For instance, if the role includes `"APPLY MARKDOWN"` in its description, the output for this role will be Markdown-formatted. This applies to both default and custom roles. If you would like to disable Markdown formatting, edit the default role description in `~/.config/shell_gpt/roles`.

**Full Changelog**: https://github.com/TheR1D/shell_gpt/compare/1.2.0...1.3.0

<hr /><em>This discussion was created from the release <a href='https://github.com/TheR1D/shell_gpt/releases/tag/1.3.0'>1.3.0</a>.</em>

**Comment by jrknox1977** [comment]
Farkhod, Amazing work as always!!! Thank you for your work on this project!!

**Comment by NikitaMGrimm** [comment]
Thank you. Are LiteLLM function calls using together.ai supported? They were added into LiteLLM a week ago. I tried it with the preinstalled example `ls` function but it looks like the parameter is being rejected.
